File: eTravelAgencija.Model/model/ReservationPreview.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace eTravelAgencija.Model.model
{
   public partial class ReservationPreview
   {
      public string OfferTitle { get; set; }
      public string HotelTitle { get; set; }
      public string HotelStars { get; set; }
      public string RoomType { get; set; }
      public decimal BasePrice { get; set; }
      public decimal Insurance { get; set; }
      public decimal ResidenceTaxTotal { get; set; }
      public decimal TotalPrice { get; set; }
      public bool IncludeInsurance { get; set; }
      public string VoucherCode { get; set; }
   }

}

File: UI/etravel_mobile/lib/models/reservation_preview.dart
import 'package:json_annotation/json_annotation.dart';

part 'reservation_preview.g.dart';

@JsonSerializable()
class ReservationPreview {

  final String offerTitle;
  final String hotelTitle;
  final String hotelStars;
  final String roomType;
  final double basePrice;
  final double insurance;
  final double residenceTaxTotal;
  final double totalPrice;

  final bool includeInsurance;
  final String? voucherCode;

  ReservationPreview({
    required this.offerTitle,
    required this.hotelTitle,
    required this.hotelStars,
    required this.roomType,
    required this.basePrice,
    required this.insurance,
    required this.residenceTaxTotal,
    required this.totalPrice,
    required this.includeInsurance,
    this.voucherCode,
  });

  factory ReservationPreview.fromJson(Map<String, dynamic> json) =>
      _$ReservationPreviewFromJson(json);

  Map<String, dynamic> toJson() => _$ReservationPreviewToJson(this);
}
